Realm/RLMRealm.mm

```diff
@@ -281,21 +242,7 @@ + (RLMAsyncOpenTask *)asyncOpenWithConfiguration:(RLMRealmConfiguration *)config
             };
 
             auto realm = Realm::get_shared_realm(std::move(ref));
-            bool needsSubscriptions = realm->is_partial() && ObjectStore::table_for_object_type(realm->read_group(), "__ResultSets")->size() == 0;
-            if (needsSubscriptions) {
-                // We need to dispatch back to the work queue to wait for the
-                // subscriptions as we're currently running on the sync worker
-                // thread and blocking it to wait for subscriptions means no syncing
-                dispatch_async(s_async_open_queue, ^{
-                    @autoreleasepool {
-                        waitForPartialSyncSubscriptions(realm->config());
-                        complete();
-                    }
-                });
-            }
-            else {
-                complete();
-            }
+            complete();
```

Comment:
The `auto realm = Realm::get_shared_realm(std::move(ref));` line above should also go away and this block should just be inlined since it's no longer called from multiple places.